### PR TITLE
Add certificate callback.

### PR DESCRIPTION
**Pull request by alexchamberlain**
Existing unit tests pass. Adapted from `__init__.py`.

### CODE REVIEW

File: test/test_remote.py [hunk 1]
```diff
@@ -242,6 +242,16 @@ def test_fetch(self):
         self.assertEqual(stats.indexed_objects, REMOTE_REPO_OBJECTS)
         self.assertEqual(stats.received_objects, REMOTE_REPO_OBJECTS)
 
+    def test_fetch_insecure(self):
+        def no_check(certificate, valid, host):
+            return True
+        remote = self.repo.remotes[0]
+        remote.certificate_check = no_check
```
Comment (alexchamberlain) [inline test/test_remote.py]: Hmmm this isn't being called, as the remote isn't `https`, but there don't appear to be any `https` tests?

Comment (carlosmn) [inline test/test_remote.py]: There's a few remotes which do use HTTPS, but this test needs to make sure it's using HTTPS, not assume that some unspecified remote is going to be that way.

Comment (alexchamberlain) [inline test/test_remote.py]: Sorry; missed this update. Any idea which test case uses HTTPS?

Comment (carlosmn) [inline test/test_remote.py]: Most should get redirected automatically through GitHub, though some credential tests use it explicitly. That shouldn't matter too much. If you need HTTPS in your test, use a HTTPS url, you're creating a new test, you don't need to use an existing url or remote.

**Comment by jdavid** [comment]
Which is the state of this pull-request?

**Comment by webknjaz** [comment]
@alexchamberlain any updates?

**Comment by jdavid** [comment]
This was implemented in commit 563cb901, just 3 months after this PR was created. Now in commit db3d3990 I've added a unit test.

**Pull request closed without merging**